src/btree/bt_vrfy.c

```diff
@@ -709,22 +709,31 @@ __verify_row_int_key_order(
 
     btree = S2BT(session);
 
-    /* The maximum key is set, we updated it from a leaf page first. */
-    WT_ASSERT(session, vs->max_addr->size != 0);
+    /*
+     * The maximum key is usually set from the leaf page first. If the first leaf page is corrupted,
+     * it is possible that the key is not set. In that case skip this check.
+     */
+    if (!vs->read_corrupt)
+        WT_ASSERT(session, vs->max_addr->size != 0);
```

Comment:
This means we will skip the ASSERT test any time we are running with the `read_corrupt` option. Can we narrow this so we only skip the test if we have seen an error? It looks like `vs->verify_err` would tell us that, I think.

Reply:
good point! changed it [38c73b2](https://github.com/wiredtiger/wiredtiger/pull/9209/commits/38c73b2e790c227e17c5c66f2d20bdfbbfa2cec3)